src/main/java/be/seeseemelk/mockbukkit/MockBukkit.java

```diff
@@ -44,20 +44,34 @@ protected static void setServerInstanceToNull()
 	 * @return The created {@link ServerMock}.
 	 */
 	public static ServerMock mock()
+	{
+		return mock(new ServerMock());
+	}
+
+	/**
+	 * Start mocking the <code>Bukkit</code> singleton.
+	 * You can pass your own implementation of the {@link ServerMock} instance.
+	 * The instance you passed is returned.
+	 *
+	 * @param serverMockImplementation your custom {@link ServerMock} implementation.
+	 * @return The provided {@link ServerMock}.
+	 */
+	@SuppressWarnings("unchecked")
+	public static <TServerMock extends ServerMock> TServerMock mock(TServerMock serverMockImplementation)
 	{
 		if (mock != null)
 		{
 			throw new IllegalStateException("Already mocking");
 		}
-		
-		mock = new ServerMock();
-		
+
+		mock = serverMockImplementation;
+
 		Level defaultLevel = mock.getLogger().getLevel();
 		mock.getLogger().setLevel(Level.WARNING);
 		Bukkit.setServer(mock);
 		mock.getLogger().setLevel(defaultLevel);
-		
-		return mock;
+
+		return (TServerMock) mock;
```

Comment:
You could also just return "serverMockImplementation" here since its the same object. This would also allow you to remove this unnecessary cast and the warnings supressor. Other than that I find this to be a good idea and hope this gets merged!

Reply:
Yeah thanks for catching that. I updated the code.